src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -0,0 +1,177 @@
+"""
+File       : MSPileupData.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileupData class provides whole logic behind
+data used and stored by MSPileup module. The data representation is the following:
+
+{
+    "pileupName": string with the pileup dataset
+    "pileupType": string with a constant value
+    "insertTime": string human readable date/time of pileup object was added to the system
+    "lastUpdateTime": string human readable date/time of pileup object was last modified by an user
+    "expectedRSEs": ["Disk1", "Disk2", etc],  # these values need to be validated against Rucio
+    "currentRSEs": ["Disk1", "Disk3"],  # values provided by the MS itself
+    "fullReplicas": integer,  # total number of replicas to keep on Disk
+    "active": boolean,
+    "pileupSizeGB": integer, current size of the pileup in GB (integer)
+    "rulesList: list of strings (rules) used to lock the pileup id
+}
+
+The data flow should be done via list of objects, e.g.
+[<pileup object 1>, <pileup object 2>, ..., <pileup object n>]
+"""
+
+# third party modules
+from pymongo import IndexModel, errors
+
+# WMCore modules
+from WMCore.Database.MongoDB import MongoDB
+from WMCore.MicroService.MSCore.MSCore import MSCore
+
+
+def stripKeys(docs, stripKeys=['_id']):
```

Comment:
I suppose this logic here is needed in order to strip the `_id` field from the Mongo documents. This has been already addressed in other way in the rest of the microservices using Mongo DB by explicitely defining a document schema in which the `_id` is present, hence it is preserved in both Mongo docmuemnt  and the python interpretation of the document. We better think of sharing logic.

Reply:
So far, stripKeys exist in other MS services but those are internal service API. I agree it is common function and most likely it should go into MongoDB module and called from there. I'll be happy to put it into MongoDB module though if you agree.

Reply:
Yes indeed, not a bad idea for this piece to go into the MongoDB module wrapper itself. 

I can admit, in the past it was done in at least two different ways, for two different modules. A quick lookup can lead us to: 
https://github.com/dmwm/WMCore/blob/479146e4ae1a73a91f44b9ab0ea8fe36195a2dae/src/python/WMCore/MicroService/MSOutput/MSOutput.py#L959-L968 
and 
https://github.com/dmwm/WMCore/blob/479146e4ae1a73a91f44b9ab0ea8fe36195a2dae/src/python/WMCore/MicroService/MSUnmerged/MSUnmergedRSE.py#L72-L90

They are addressing some slightly different use cases. And I am more in favor of  the later, since it is more friendly to the model where we have a strictly defined docSchema. And the projection can be put directly into the MongoDB query. But this one was also covering some subtle details in MSUnmerged, so we need to be careful if we decide to go for unification of this method.